### DISCUSSION TITLE
Auto Apply with browser dark mode

### DISCUSSION THREAD

**Discussion by SamYuan1990** [anuraghazra/github-readme-stats#1684] [Q&A] [answered]
Hello everyone,

As you know we have dark and light model in browser.
and I am not sure if we are able to auto apply browser setting with this project when usage.

so that we can avoid hard code light/dark in text code and make it auto apply.

Thanks
Sam

**Comment by rickstaa** [comment, marked as answer]
@SamYuan1990 Thanks for your question. This feature has been discussed in multiple issues and discussions:
- #1645
- #1182
- #897
- #777
- #739
- #1036 
- #689
- #414
- #79
- #693
- #1415 
- #690
- #1880
- #1684
- #1920
 
For now, people are advised to use [this trick](https://github.com/anuraghazra/github-readme-stats/discussions/1415#discussioncomment-2152317)

**Reply by SamYuan1990** [reply]
# [![Readme Card](https://github-readme-stats.vercel.app/api/pin/?username=hyperledger-labs&repo=PerformanceSandBox&theme=tokyonight#gh-dark-mode-only)](https://github.com/hyperledger-labs/PerformanceSandBox)
# [![Readme Card](https://github-readme-stats.vercel.app/api/pin/?username=hyperledger-labs&repo=PerformanceSandBox&theme=default#gh-light-mode-only)](https://github.com/hyperledger-labs/PerformanceSandBox)

**Reply by SamYuan1990** [reply]
@rickstaa , it seems won't work for `Readme Card` for now?

**Reply by SamYuan1990** [reply]
but it works when I use vs-code to do the review...

**Reply by SamYuan1990** [reply]
I see... it is strange for me, at git hub it finally works with

# [![Readme Card](https://github-readme-stats.vercel.app/api/pin/?username=hyperledger-labs&repo=PerformanceSandBox&theme=tokyonight#gh-dark-mode-only)](https://github.com/hyperledger-labs/PerformanceSandBox#gh-dark-mode-only)
# [![Readme Card](https://github-readme-stats.vercel.app/api/pin/?username=hyperledger-labs&repo=PerformanceSandBox&theme=default#gh-light-mode-only)](https://github.com/hyperledger-labs/PerformanceSandBox#gh-light-mode-only)

which means on github markdown, we need `#gh-light-mode-only` appears at link url as identifier? but not from the picture source?

**Reply by rickstaa** [reply]
@SamYuan1990 Sorry for the late reply. I have not used this trick myself so I had to investigate it looks like the right syntax is:

```markdown
[![Readme Card](https://github-readme-stats.vercel.app/api/pin/?username=hyperledger-labs&repo=PerformanceSandBox&theme=dark#gh-dark-mode-only)](https://github.com/hyperledger-labs/PerformanceSandBox#gh-dark-mode-only)
[![Readme Card](https://github-readme-stats.vercel.app/api/pin/?username=hyperledger-labs&repo=PerformanceSandBox&theme=default#gh-light-mode-only)](https://github.com/hyperledger-labs/PerformanceSandBox#gh-light-mode-only)
````

or

```html
<a href="https://github.com/itaditya#gh-dark-mode-only">
  <img src="https://github-readme-stats.vercel.app/api/pin/?username=hyperledger-labs&repo=PerformanceSandBox&theme=dark#gh-dark-mode-only" alt="My GitHub Stats" />
</a>
<a href="https://github.com/itaditya#gh-light-mode-only">
  <img src="https://github-readme-stats.vercel.app/api/pin/?username=hyperledger-labs&repo=PerformanceSandBox&theme=default#gh-light-mode-only" alt="My GitHub Stats" />
</a>
```
[![Readme Card](https://github-readme-stats.vercel.app/api/pin/?username=hyperledger-labs&repo=PerformanceSandBox&theme=dark#gh-dark-mode-only)](https://github.com/hyperledger-labs/PerformanceSandBox#gh-dark-mode-only)
[![Readme Card](https://github-readme-stats.vercel.app/api/pin/?username=hyperledger-labs&repo=PerformanceSandBox&theme=default#gh-light-mode-only)](https://github.com/hyperledger-labs/PerformanceSandBox#gh-light-mode-only)

Using the following syntax seems to fail:

```markdown
![Readme Card](https://github-readme-stats.vercel.app/api/pin/?username=hyperledger-labs&repo=PerformanceSandBox&theme=dark#gh-dark-mode-only)
![Readme Card](https://github-readme-stats.vercel.app/api/pin/?username=hyperledger-labs&repo=PerformanceSandBox&theme=default#gh-light-mode-only)
```

![Readme Card](https://github-readme-stats.vercel.app/api/pin/?username=hyperledger-labs&repo=PerformanceSandBox&theme=dark#gh-dark-mode-only)
![Readme Card](https://github-readme-stats.vercel.app/api/pin/?username=hyperledger-labs&repo=PerformanceSandBox&theme=default#gh-light-mode-only)

**Reply by erikw** [reply]
Thanks @rickstaa, I can confirm that this syntax works:

```markdown
[![Readme Card](https://github-readme-stats.vercel.app/api/pin/?username=hyperledger-labs&repo=PerformanceSandBox&theme=dark#gh-dark-mode-only)](https://github.com/hyperledger-labs/PerformanceSandBox#gh-dark-mode-only)
[![Readme Card](https://github-readme-stats.vercel.app/api/pin/?username=hyperledger-labs&repo=PerformanceSandBox&theme=default#gh-light-mode-only)](https://github.com/hyperledger-labs/PerformanceSandBox#gh-light-mode-only)
```

@anuraghazra could we add these syntax instructions to the project README, to make it easier for folks to use pinned repo cards with automatic light/dark mode toggle? :)

**Comment by rickstaa** [comment]
As of today, this also is possible using the HTML `<picture>` element in combination with the `prefers-color-scheme` media feature. It is also (for more information, see https://github.blog/changelog/2022-05-19-specify-theme-context-for-images-in-markdown-beta/). Please, however, be aware that in contrast to the method above, this method does not work when people switch the theme without reloading the page.

**Reply by SamYuan1990** [reply]
I see.

**Reply by rickstaa** [reply]
Here is an example:


```html
<a href="https://github.com/anuraghazra/github-readme-stats">
<picture>
  <source media="(prefers-color-scheme: dark)" srcset="https://github-readme-stats.vercel.app/api?username=anuraghazra&theme=dark">
  <img alt="Shows Anuraghazra's GitHub Stats." src="https://github-readme-stats.vercel.app/api?username=anuraghazra&theme=default">
</picture>
</a>
```

<a href="https://github.com/anuraghazra/github-readme-stats">
<picture>
  <source media="(prefers-color-scheme: dark)" srcset="https://github-readme-stats.vercel.app/api?username=anuraghazra&theme=dark">
  <img alt="Shows Anuraghazra's GitHub Stats." src="https://github-readme-stats.vercel.app/api?username=anuraghazra&theme=default">
</picture>
</a>

**Comment by rickstaa** [comment]
All the currently available ways for creating a dynamic theme are now documented in the [README.md](https://github.com/anuraghazra/github-readme-stats#responsive-card-theme) (See https://github.com/anuraghazra/github-readme-stats#responsive-card-theme).
